protocols/virtualfund/virtualfund.go

```diff
@@ -30,9 +30,18 @@ const ObjectivePrefix = "VirtualFund-"
 // errors
 var ErrNotApproved = errors.New("objective not approved")
 
+// GuaranteeInfo contains the information used to generate the expected guarantees.
+type GuaranteeInfo struct {
+	Left                 types.Destination
+	Right                types.Destination
+	LeftAmount           types.Funds
+	RightAmount          types.Funds
+	GuaranteeDestination types.Destination
+}
 type Connection struct {
 	Channel            *channel.TwoPartyLedger
 	ExpectedGuarantees map[types.Address]outcome.Allocation
+	GuaranteeInfo      GuaranteeInfo
```

Comment:
`ExpectedGuarantees` contains most of the information in `GuaranteeInfo` but not all of it. It would contain the total amount of funding for the guarantee but it doesn't specify how much of that is `LeftAmount` and how much is `RightAmount` since the guarantee only has the [total](https://github.com/statechannels/go-nitro/blob/055b983b6b5e5c20f558c151cd3bb869b1d99893/protocols/virtualfund/virtualfund.go#L491).

I think we could calculate `ExpectedGuarantees` from `GuaranteeInfo` if we wanted to though. If we're interested in doing that, it might make sense to do that in a separate PR as `ExpectedGuarantees` is used in a bunch of our tests.

Reply:
in that PR, it would be nifty if `Connection` implemented the `ExpectedGuarantees()` method!

another alternative is to modify `ExpectedGuarantees` to include the `LeftAmount` and `RightAmount`. In that way, the actual guarantees (the data structure included in a channel's outcome) are easily computable, by summing `LeftAmount` and `RightAmount`.